src/test/java/test/DynamicGraphTest.java

```diff
@@ -133,6 +140,55 @@ public void test2() {
   }
 
   @Test
+  public void test3() {
+          DynamicGraph<Node> dg = new DynamicGraph<>();
+          Node a = new Node("a");
+          Node b = new Node("b");
+          Node c = new Node("c");
+          dg.addNode(a);
+          dg.addNode(b);
+          dg.addNode(c);
+          dg.addEdge(1, a, b);
+          dg.addEdge(0, c, b);
+          dg.addEdge(0, b, a);
+
+          assertFreeNodesEquals(dg, b);
+
+          dg.setStatus(b, Status.RUNNING);
+          assertFreeNodesEquals(dg);
+
+          dg.setStatus(b, Status.FINISHED);
+          assertFreeNodesEquals(dg, a, c);
```

Comment:
This also contains circular references which causes the confusion. The addition of this test (test3) probably doesn't really add anything and perhaps should be removed. I think test4 is the only valid test in this class aside from testOrderingOfEdgesWithSameWeight as all the others have cyclic dependencies.